app/controllers/application_controller.rb

```diff
@@ -14,41 +14,33 @@ def invalid_request(message = error_msg, status = 400)
     render json: {errors: message}, status: status
   end
 
-  def login
-    user = authenticate_cookie
-    @current_user = User.from_andela_auth(user) if user && !user['isGuest']
-    return unauthorized_token unless @current_user
-
-    @token = TokenManager.generate_token(@current_user.id)
-  end
-
-  def logout
-    cookie = request.headers['HTTP_ANDELA_COOKIE']
-    return invalid_request("Cookie must be provided") unless cookie
-    CookieHandler.logout_cookie(cookie)
-    render json: {message: 'logged out'}, status: 200
-  end
-
 private
-  def authenticate_user
-    (authenticate_token && authenticate_cookie) || unauthorized_token
-    # authenticate_cookie || unauthorized_token
-  end
-
-  def authenticate_token
-    authenticate_with_http_token do |auth_token, _|
-      user_id = TokenManager.authenticate(auth_token)['user']
-      @token = TokenManager.generate_token(user_id) if user_id
+   def authenticate_user
```

Comment:
Not a blocker, but please refactor this method!
I'm thinking you should be able to do something like: 
```ruby
def authenticate_user
  auth = Authenticator.new(request) # pass the request object to the authenticator, you can fetch the details from the request object, i.e let the authenticator determine how it want's to handle it's auth, either by params or header
  # using this approach, you're able to switch authenticators easily! Authenticator only becomes an interface for authentication, that defines 2 public methods: `authenticated?` and `user`.
  # this means when a new version of authentication is to be used, you don't have to touch this controller at all. Also authentication becomes a bit more modular!
  if auth.authenticated?
    # you don't have to define the `create_or_update_user` in the controller
    create_or_update_user(auth.user)
  else
    unauthorized_token
  end
end
```